polygeist/tools/cgeist/driver.cc

```diff
@@ -855,32 +836,75 @@ static int compileModule(mlir::OwningOpRef<mlir::ModuleOp> &module,
   return 0;
 }
 
-static void setMLIROptLevel(llvm::StringRef Arg) {
-  switch (Arg[2]) {
-  case '0':
-    Opt0 = true;
-    break;
-  case '1':
-    Opt1 = true;
-    break;
-  case '2':
-    Opt2 = true;
-    break;
-  case '3':
-    Opt3 = true;
-    break;
+static llvm::OptimizationLevel getOptimizationLevel(unsigned SpeedLevel) {
+  switch (SpeedLevel) {
+  case 0:
+    return llvm::OptimizationLevel::O0;
+  case 1:
+    return llvm::OptimizationLevel::O1;
+  case 2:
+    return llvm::OptimizationLevel::O2;
   default:
-    llvm::WithColor::warning()
-        << "Optimization level " << Arg << " not handled by cgeist\n";
+    // All speed levels above 2 are equivalent to '-O3'
+    llvm::WithColor::warning() << "optimization level '-O" << SpeedLevel
+                               << "' is not supported; using '-O3' instead\n";
+    LLVM_FALLTHROUGH;
+  case 4:
+    // '-O4' warning is displayed before reaching this point
+  case 3:
+    return llvm::OptimizationLevel::O3;
   }
 }
 
+static constexpr llvm::OptimizationLevel getOptimizationLevel(char SizeLevel) {
+  switch (SizeLevel) {
+  case 's':
+    return llvm::OptimizationLevel::Os;
+  case 'z':
+    return llvm::OptimizationLevel::Oz;
+  case 'g':
+    // '-Og' is equivalent to '-O1'
+    return llvm::OptimizationLevel::O1;
+  default:
+    llvm_unreachable("Invalid optimization level");
+  }
+}
+
+static llvm::OptimizationLevel parseOptimizationLevel(llvm::StringRef Arg) {
+  if (Arg.empty()) {
+    // -O and --optimize options are equivalent to -O1
+    return llvm::OptimizationLevel::O1;
+  }
+
+  if (Arg.startswith("fast")) {
+    // We handle -Ofast like -O3.
```

Comment:
Do we really want to handle `-0fast` the same as `-O3`? 

`-Ofast` enables more optimizations that might be relevant for the user.

Reply:
When compiling, we'll pass the `-Ofast` flag itself, but a default pipeline for `-Ofast` is not available.

Reply:
What do you mean by "compiling" here? Compiling the LLVM IR obtained from MLIR to object?

Reply:
Yes, that's what I meant.